### PR TITLE
Refreshing JSI interfaces to latest

### PR DESCRIPTION
**Pull request by mganandraj**
This change will is needed once the latest JSI interfaces from the facebook react-native/Hermes repo are brough to Microsoft fork. Essentially, the fixes takes into account the minor change in the JSI header location and the added methods in it.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2850)

### CODE REVIEW

**Comment by mganandraj** [comment]
> This change will is needed once the latest JSI interfaces from the facebook react-native/Hermes repo are brough to Microsoft fork. Essentially, the fixes takes into account the minor change in the JSI header location and the added methods in it.
> Microsoft Reviewers: Open in CodeFlow

This will enabled us to integrate Hermes without waiting for the Microsoft fork of react-native getting uptodate with facebook repo.

File: vnext/ReactCommon/ReactCommon.vcxproj [hunk 1]
```diff
@@ -137,8 +137,10 @@
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\JSCRuntime.cpp">
       <ExcludedFromBuild>true</ExcludedFromBuild>
     </ClCompile>
-    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi.cpp" />
-    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\JSIDynamic.cpp" />
+    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi\jsi.cpp" />
+    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi\JSIDynamic.cpp" />
+    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi\jsilib-posix.cpp" />
```
Comment (hansenyy) [inline vnext/ReactCommon/ReactCommon.vcxproj]: ><ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi\jsilib-posix.cpp" /> [](start = 4, length = 78)

Why do we need this file to be compiled as part of the Windows build?

File: vnext/ReactCommon/ReactCommon.vcxproj [hunk 2]
```diff
@@ -137,8 +137,10 @@
     <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\JSCRuntime.cpp">
       <ExcludedFromBuild>true</ExcludedFromBuild>
     </ClCompile>
-    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi.cpp" />
-    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\JSIDynamic.cpp" />
+    <ClCompile Include="$(ReactNativeDir)\ReactCommon\jsi\jsi\jsi.cpp" />
```
Comment (jonthysell) [inline vnext/ReactCommon/ReactCommon.vcxproj]: Be sure to add any new entries to ReactCommon.vcxproj.filters

File: vnext/ReactCommon/ReactCommon.vcxproj (hunk 1 above)
Comment (mganandraj) [reply vnext/ReactCommon/ReactCommon.vcxproj]: It had the macro blocks hence included .. but removed it now.. Thanks!!

File: vnext/ReactCommon/ReactCommon.vcxproj (hunk 2 above)
Comment (mganandraj) [reply vnext/ReactCommon/ReactCommon.vcxproj]: Done.. in the new PR. I have to revert this one as i can't push updates to this branch.

**Pull request closed without merging**